### PR TITLE
Adding period (from-to) to nearby nodes and people

### LINKED ISSUE #4558: Nearby nodes and people
Nearby nodes: `/api/srch/taglocations` (`tagNearbyNodes()` method)
Nearby people: `/api/srch/nearbyPeople` (`tagNearbyPeople()` method)

>- "Brainstorm necessary & useful default and settable sorting for content and people with an “order” parameter"
> - "Explore time bounding parameters for display over “past week” or “past month” or given period consider ways to group people by topic focus" @jywarren

Tasks:
* [x] (#4597) Add "order" parameter to `taglocations`: sort the nodes by date of creation, updates and oldest. / Add more options to "order"  parameter to `nearbyPeople`: Most nodes posted and signup date.
* [ ] (#4604) Add "period" parameter to `taglocations`: add 'start' and 'end' parameters and the search would return results only for that specific period. End default should be now. / Add "period" parameter to `nearbyPeople`: add 'start' and 'end' parameters and the search would return only users with some activity in that specific period. It would be useful for #4560.

### PR DESCRIPTION
**Pull request by milaaraujo**
Fixes #4558

Adding options to `/api/srch/taglocations`:
- `from=[date]`: It accepts a date, if not specified (1990,01,01). It searches for nodes created from the date.
- `to=[date]`: Accepts a date, if not specified uses `now`. It searches for nodes created by the specified date.

Adding options to `/api/srch/nearbyPeople`:
- `from=[date]`: It accepts a date. It searches for users with some activity from the specified date.
- `to=[date]`: Accepts a date. It searches for users with some activity by the specified date.

### CODE REVIEW

**Comment by milaaraujo** [comment]
Hey @jywarren, can you take a look at this?

**Comment by milaaraujo** [comment]
I don't thin this error is something I did:
```
ERROR["test_username_should_not_be_updated", #<Minitest::Reporters::Suite:0x000055b9d3246a98 @name="UserTest">, 53.27949030800005]
 test_username_should_not_be_updated#UserTest (53.28s)
Authlogic::Session::Activation::NotActivatedError:         Authlogic::Session::Activation::NotActivatedError: You must activate the Authlogic::Session::Base.controller with a controller object before creating objects
            test/unit/user_test.rb:261:in `block in <class:UserTest>'
```

**Comment by jywarren** [comment]
whoa, are you rebased over the latest master branch? If so, could you open
an issue for this? Thanks for reporting it in!

**Comment by gauravano** [comment]
Hi, this is weird as this test is added today only and  passed the test too - https://github.com/publiclab/plots2/pull/4599/files . Retriggering the build for this PR. Thanks!

File: app/api/srch/shared_params.rb [hunk 1]
```diff
@@ -31,5 +31,10 @@ module SharedParams
     params :field do
       optional :field, type: String, documentation: { example: 'username' }
     end
+
+    params :period do
+      optional :from, type: Date, documentation: { example: '14/12/1989' }
```
Comment (jywarren) [inline app/api/srch/shared_params.rb]: Here, I think @cesswairimu is using a `-` hyphen convention in other date code. Could we switch to that, do you think?

Comment (milaaraujo) [reply app/api/srch/shared_params.rb]: Sure! And what validation are you talking about? `search_criteria.validate_period_from_to`?

**Comment by milaaraujo** [comment]
Hey Jeff, it is rebased now. Can you help me with that error? Not sure what is the problem!

**Comment by milaaraujo** [comment]
Problem fixed!

File: app/api/srch/search.rb [hunk 2]
```diff
@@ -348,7 +348,7 @@ class Search < Grape::API
     def self.execute(endpoint, params)
```
Comment (jywarren) [inline app/api/srch/search.rb]: Yes for validation maybe just passing some bad data into execute() and confirming that it rejects it.

**Comment by milaaraujo** [comment]
After a good fight with codeclimate, I think it is ready!

File: test/functional/api/search_api_test.rb [hunk 3]
```diff
@@ -118,8 +118,91 @@ def app
...
+
+    assert matcher    =~ json
+
+    assert_equal 23,  json['items'][0]['doc_id']
+    assert_equal 24,  json['items'][1]['doc_id']
+    assert_equal 25,  json['items'][2]['doc_id']
+  end
+
+  test 'search Tag Nearby Nodes functionality with a valid query and from date greater then to date' do
+    get '/api/srch/taglocations?nwlat=201.0&selat=0.0&nwlng=0.0&selng=200.0&sort_by=recent&order_direction=ASC&from=2018-08-16&to=2018-07-31'
```
Comment (jywarren) [inline test/functional/api/search_api_test.rb]: Nice!

File: test/functional/api/search_api_test.rb [hunk 4]
```diff
@@ -146,8 +229,28 @@ def app
     assert matcher =~ json
   end
 
+  test 'search Tag Nearby People functionality with wrong query' do
+    get '/api/srch/nearbyPeople?nwlat=31.0&selat=0.0&nwlng=0.0&selng=40.0&from=date'
```
Comment (jywarren) [inline test/functional/api/search_api_test.rb]: Great!

File: test/functional/api/search_api_test.rb [hunk 5]
```diff
@@ -118,8 +118,91 @@ def app
...
+        }.ignore_extra_keys!
+    }.ignore_extra_keys!
+
+    matcher = JsonExpressions::Matcher.new(pattern)
+    json = JSON.parse(last_response.body)
+
+    assert_equal "from is invalid", json["error"]
+  end
+
+  test 'search Tag Nearby Nodes functionality with a valid query and specific period' do
```
Comment (jywarren) [inline test/functional/api/search_api_test.rb]: 🎉

**Comment by jywarren** [comment]
I don't completely know what happened here, @milaaraujo, but it looks like a few items got merged in from other commits, and weren't cleanly rebased. Not a big deal, but maybe a true clean rebase next time would help keep this PR's commits focused more narrowly. Just to watch out for, thanks!

**Comment by milaaraujo** [comment]
I'm not really good with `git rebase`! :sob: Next time I will be more careful. Thanks

**Comment by jywarren** [comment]
:-) thanks!

**Pull request merged**